File: python practice/20260613_Inception.py
import sys

def main():
    input_data = sys.stdin.read().split()
    if not input_data:
        return
    
    n = int(input_data[0])
    idx = 1
    stack = []
    
    for _ in range(n):
        if idx >= len(input_data):
            break
        op = input_data[idx]
        if op == "Sleep":
            stack.append(input_data[idx+1])
            idx += 2
        elif op == "Kick":
            if stack:
                stack.pop()
            idx += 1
        elif op == "Test":
            print(stack[-1] if stack else "Not in a dream")
            idx += 1

if __name__ == '__main__':
    main()

File: C Language/20260613_Inception.c
#include <stdio.h>
#include <string.h>

char stack[50000][20];
int top = -1;

int main() {
    int n;
    if (scanf("%d", &n) != 1) return 0;

    char op[10], name[20];
    for (int i = 0; i < n; i++) {
        scanf("%s", op);
        if (strcmp(op, "Sleep") == 0) {
            scanf("%s", name);
            strcpy(stack[++top], name);
        } else if (strcmp(op, "Kick") == 0) {
            if (top >= 0) top--;
        } else if (strcmp(op, "Test") == 0) {
            if (top >= 0) printf("%s\n", stack[top]);
            else printf("Not in a dream\n");
        }
    }
    
    return 0;
}